### ISSUE TITLE
Installing over an existing/running pdns server

### ISSUE THREAD

**Issue by sriccio** [loewexy/pdnsmanager#97]
Hello,

I'm trying to run pdnsmanger 2.01 over the master host of an existing/running pdns cluster (native with mysql replication).

Everything is find until I give the setup the database in the installer.

It says : Database is not empty.

I found another topic about an older version of pdnsmanger that makes me think using an existing database is supported or at least was in 1.x. https://github.com/loewexy/pdnsmanager/issues/64

Am I maybe doing something wrong ?

**Comment by sriccio** [comment]
I found in the setup controller that this is always checked.


`            // Check if database is empty
            $query = $pdo->prepare('SHOW TABLES');
            $query->execute();
            if ($query->fetch() !== false) {
                return $res->withJson(['error' => 'The database is not empty.'], 500);
            }
`

It would really be nice that using an existing db is supported. We don't really want to wipe our 15k powerdns domains database.

Why is an empty database needed ? Isn't the database schema the same except maybe some added tables ?

**Comment by loewexy** [comment]
It seems to complicated to make the installer work with any arbitrary existing database.

But you can check in backend/src/sql there is the SQL file used for setup. If you compare this to your database and adjust it accordingly you can setup PDNS Manager by manually creating the config file. Also then you have to add a user to the database manually or use the config plugin https://pdnsmanager.org/documentation/configuration_authentication/#config-plugin

The other way would be to install PDNS Manager to an empty database, and import the data from the old database into the tables.

**Comment by megabert** [comment]
Hi Sébastien,

I'm trying to do the same here. I did not want to too, but it seems I have to have an additional interface, because powerdns-admin has serious bug, which I must find a way around: 
https://github.com/ngoduykhanh/PowerDNS-Admin/issues/479

So I'm starting to figure out how it could be done. Maybe this thread is a good place to talk about the things we found out about how to get it done. Use every information here at your own risk! Have enough backups! Ideally try in a testing environment first!

At first please note my other issue #98 about possible different data types for a constraint.

This is the sql-file I used to extend my schema(powerdns 4.1):

```
CREATE TABLE `permissions` (
          `domain_id` int(11) NOT NULL,
          `user_id` int(11) NOT NULL,
          PRIMARY KEY (`domain_id`, `user_id`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8;

CREATE TABLE `remote` (
          `id` int(11) NOT NULL AUTO_INCREMENT,
          `record` bigint(20) NOT NULL,
          `description` varchar(255) NOT NULL,
          `type` varchar(20) NOT NULL,
          `security` varchar(2000) NOT NULL,
          `nonce` varchar(255) DEFAULT NULL,
          PRIMARY KEY (`id`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8;

CREATE TABLE `users` (
          `id` int(11) NOT NULL AUTO_INCREMENT,
          `name` varchar(50) NOT NULL,
          `backend` varchar(50) NOT NULL,
          `type` varchar(20) NOT NULL,
          `password` varchar(255) DEFAULT NULL,
          PRIMARY KEY(`id`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8;

CREATE TABLE `options` (
          `name` varchar(255) NOT NULL,
          `value` varchar(2000) DEFAULT NULL,
          PRIMARY KEY (`name`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8;

INSERT INTO `options` (`name`, `value`) VALUES
('schema_version', '5');

ALTER TABLE `permissions`
  ADD CONSTRAINT `permissions_ibfk_1` FOREIGN KEY (`domain_id`) REFERENCES `domains` (`id`) ON DELETE CASCADE,
  ADD CONSTRAINT `permissions_ibfk_2` FOREIGN KEY (`user_id`) REFERENCES `users` (`id`) ON DELETE CASCADE;

ALTER TABLE `remote`
  ADD CONSTRAINT `remote_ibfk_1` FOREIGN KEY (`record`) REFERENCES `records` (`id`) ON DELETE CASCADE;

UPDATE users SET backend='native' WHERE backend='';

UPDATE options SET value=6 WHERE name='schema_version';```

**Comment by sriccio** [comment]
Hello @megabert,

Thanks a lot for your script. Unfortunately I saw it too late.

I was going to write back to this issue that I was able to install it over our existing database. 

I used the lazy mode and  downloaded Navicat Premium, opened both database and did a schema compare, selected what changes I wanted to apply from the pdnsmanger created db to our existing db.

This went smoother than I've excepted, pdnsmanager is now running on our pdns db.

ps: I also installed pdnsmanager to replace powerdns-admin which is actually too slow/buggy with large pdns databases.
